test-case/verify-ucm-config.sh

```diff
@@ -0,0 +1,123 @@
+            ret=`alsaucm -c "$cardname" set _verb "$verb" set _enadev "$device" list1 _enadevs`
+            [[ $? -ne 0 ]] && dloge "enable '$verb':'$device' failed" && IFS="$OLD_IFS" && return 1
+            ret=`echo $ret |grep "$device" 2>/dev/null`
+            [[ -z "$ret" ]] && dloge "'$verb':'$device' is not in the enabled device list" && IFS="$OLD_IFS" && return 1
+
+	    # enable the device and setup the controls to check if the controls can be set correctly or not
+            dlogc "alsaucm -c '$cardname' set _verb '$verb' set _enadev '$device' reload"
+            alsaucm -c "$cardname" set _verb "$verb" set _enadev "$device" reload
+            [[ $? -ne 0 ]] && dloge "reload '$verb':'$device' setting failed!" && IFS="$OLD_IFS" && return 1
+
+            # device format: the first line is device and the second line is description
+            # Let's skip the description
+            ((descline=1))
+        else
+            ((descline=0))
+        fi
+    done
+    IFS="$OLD_IFS"
+    return 0
+}
+
+ucm_folder=/usr/share/alsa/ucm
+
+sofcard=${SOFCARD:-0}
+cardname=$(sof-dump-status.py -s $sofcard)
+[[ $? -ne 0 ]] && exit 1
+
+# TODO: check the audio card dmi name (dmi name is not recommanded).
+ucm_file=${ucm_folder}/${cardname}/${cardname}.conf
+
+[[ ! -e "${ucm_file}" ]] && dloge "ucm configuration: $ucm_file not exists!" && exit 2
+
+# currently we only supports old ucm configuration.
+# only after alsa-lib, alsa-utils and pulseaudio are updated, UCMv2 will be supported.
+# check ucm version
+[[ "$(grep "Syntax 2" "${ucm_file}")" ]] && dloge "not support UCMv2 currently" && exit 2
+
+# use alsaucm to check the ucm configurations
+dlogc "alsaucm open '$cardname'"
+# maybe use cardname let alsaucm choose ucm file
+alsaucm open "$cardname"
+[[ $? -ne 0 ]] && dloge "open card '$cardname' failed!" && exit 1
+
+dlogc "alsaucm -c "$cardname" reload"
+alsaucm -c "$cardname" reload
+[[ $? -ne 0 ]] && dloge "card '$cardname' reload failed. Please check the card amixer setting." && exit 1
+
+dlogc "alsaucm -c '$cardname' list _verbs"
+verbs=`alsaucm -c "$cardname" list _verbs`
+[[ $? -ne 0 ]] && dloge "'$cardname' list _verbs failed!" && exit 1
+[[ -z "$verbs" ]] && dloge "'$cardname' ucm has no verbs!" && exit 1
+((i=0))
+
+OLD_IFS="$IFS"
+IFS=$'\n'
+for verb in $verbs; do
+    if [ $i -eq "0" ]; then
+        verb=`echo $verb|awk '{print $2}'`
+        [[ -z "$verb" ]] && dlogw "verb is null, skipping" && continue
+        dlogi "found verb: $verb"
+
+        dlogc "alsaucm -c '$cardname' set _verb $verb get _verb"
+        ret=`alsaucm -c "$cardname" set _verb $verb get _verb`
+        [[ $? -ne 0 ]] && dloge "set verb: $verb failed!" && IFS="$OLD_IFS" && exit 1
+        ret=`echo $ret |grep "$verb" 2>/dev/null`
+        [[ -z "$ret" ]] && dloge "'$verb' is not the selected verb" && IFS="$OLD_IFS" && exit 1
+
+        func_verify_verb_device $cardname $verb
```

Comment:
1. IFS change without export, just effect in your script, so line 106 restore can be ignore
2. IFS will effect the function, because without `local`, all variable is global, so there will conflict with `func_verify_verb_device ` define,

So you do the too many thing in your IFS, it will effect too many commands

I suggest you store the data into array, then loop the array like this [line to array](https://github.com/thesofproject/sof-test/blob/master/case-lib/hijack.sh#L27)